### PR TITLE
Add file extensions to relative paths

### LINKED ISSUE #4: Fix routing from inner pages after refresh
1. Visit https://docs.oasis.dev/general/oasis-network/network-parameters/
2. Scroll down to Testnet link
3. Click on the link.
4.  => Results in 404 page not found because it's linking (with javascript routes window.href=...) to https://docs.oasis.dev/general/oasis-network/foundation/testnet/ (notice the redundant `/oasis-network/` chunk)

This only happens when initially visiting the URL of an internal page and then navigating from there. If you commence with the landing page and then navigating through the page, you will not encounter this. Also, oasis-core and oasis-sdk are not affected by this. Also, disabling javascript fixes the issue.

Maybe check how docusaurus.io website handles this as they also use multi-sites.

### PR DESCRIPTION
**Pull request by lukaw3d**
Fixes https://github.com/oasisprotocol/docs/issues/4

### CODE REVIEW

**Review by kostko — APPROVED** [review]

**Review by tjanez — COMMENTED** [review]
Great work!

Could you also add Markdownlint check to CI? It would be cool if there is a Markdownlint GitHub action that would also annotate the errors, otherwise `yarn lintmarkdown` is OK.

Also, could you change "Auto-fix, mostly" commit message to something more concrete :slightly_smiling_face: .

**Comment by lukaw3d** [comment]
Improved commit messages

**Review by tjanez — APPROVED** [review]
Thanks for tidying up the commits!

> Could you also add Markdownlint check to CI? It would be cool if there is a Markdownlint GitHub action that would also annotate the errors, otherwise `yarn lintmarkdown` is OK.

I've opened a follow up issue for that: #29.

File: package.json [hunk 1]
```diff
@@ -12,7 +12,9 @@
...
+    "blc": "broken-link-checker --recursive --ordered --exclude https://t.me --exclude https://software.intel.com/content --exclude https://nvlpubs.nist.gov/nistpubs/SpecialPublications/NIST.SP.800-185.pdf --exclude https://support.ledger.com/hc/en-us/articles --exclude https://www.linkedin.com --exclude https://csrc.nist.gov/publications/detail/fips/180/4/final --exclude https://www.coinspeaker.com/oasis-protocol-partnership-meta/ --exclude https://www.binance.com/en/support --exclude https://www.coinbase.com --filter-level 3 http://localhost:3000/",
+    "lint-markdown": "markdownlint-cli2 'docs/**/*.md'",
+    "fmt-markdown": "markdownlint-cli2-fix 'docs/**/*.md'"
```
Comment (lukaw3d) [inline package.json]: oops, this missed links in `.mdx` files

**Comment by lukaw3d** [comment]
I also checked referenced external repos are okay, with some manual steps:
- remove .markdownlint.yml files in external repos
- add to .markdownlint-cli2.cjs
  ```js
    noInlineConfig: true,
    rules: {
        'emphasis-style': false,
        'code-block-style': false,
        'no-space-in-code': false,
        'single-title': false,
  ```
- run `yarn markdownlint-cli2-fix 'external/*/docs/**/*.md' 'external/*/docs/**/*.mdx'`

**Comment by lukaw3d** [comment]
Running above steps + `yarn markdownlint-cli2 'external/**/*.md' 'external/**/*.mdx'` found these broken links:

external/oasis-core-ledger/CHANGELOG.md:90  [File missing] [Context: "doc/versioning.md"]
external/oasis-core-ledger/CHANGELOG.md:179  [File missing] [Context: "doc/versioning.md"]
external/oasis-core/CHANGELOG.md:1326  [File missing] [Context: "SECURITY.md"]
external/oasis-core/CHANGELOG.md:4807  [File missing] [Context: "../docs/oasis-node/metrics.md"]

**Comment by lukaw3d** [comment]
I found a good approximation for detecting bad links. Find any `../../` in build folder:
`grep -o -E '\.\./\.\./.{0,50}' -r ./build; test $? -eq 1`

And even better, find all `href:"relative` in build folder:
`grep -o -E 'href:".{0,50}' -r ./build | grep -v -E 'href:"(https:|http:|mailto:|/|#)';  test $? -eq 1`

**Pull request merged**